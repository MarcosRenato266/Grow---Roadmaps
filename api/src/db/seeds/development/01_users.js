exports.seed = function(knex) {
  // Deletes ALL existing entries
  return knex('users')
    .del()
    .then(function() {
      // Inserts seed entries
      return knex('users').insert([
        {
          id: '54378ee4-51a9-4208-b3c5-dcca2f3a01e2',
          name: 'Grow',
          username: 'grow',
          email: 'grow@gmail.com',
          profilePicture: 'https://i.ytimg.com/vi/2fb-g_V-UT4/hqdefault.jpg',
          accessLevel: 0,
        },
        {
          id: '54378ee4-51a9-4208-b3c5-dcca29dr3kk5',
          name: 'Marcos Nunes',
          username: 'marcos',
          email: 'corecodde@hotmail.com',
          profilePicture: 'https://i.ytimg.com/vi/2fb-g_V-UT4/hqdefault.jpg',
          accessLevel: 1,
        },
      ]);
    });
};
