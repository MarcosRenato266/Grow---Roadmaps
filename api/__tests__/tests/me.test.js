import knexCleaner from 'knex-cleaner';
import { request } from '../utils';
import knex from '../../src/db/knex';
import { ids } from '../../src/db/seeds/test/01_users';

let user = null;

beforeAll(async () => {
  await knexCleaner.clean(knex, {
    ignoreTables: ['knex_migrations', 'knex_migrations_lock'],
  });
  await knex.seed.run();
  user = (await knex('users').where({ id: ids.userOne }))[0];
});

it('should retrieve the current logged in user', async () => {
  const context = { user };

  const query = `
    query Me {
      me {
        id
        name
        email
      }
    }
  `;

  const result = await request(query, { context });
  expect(result).toMatchSnapshot();
});
