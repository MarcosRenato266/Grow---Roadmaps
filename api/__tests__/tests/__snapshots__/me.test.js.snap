// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should retrieve the current logged in user 1`] = `
Object {
  "data": Object {
    "me": Object {
      "email": "one@user.com",
      "id": "45ecf361-3786-494a-b299-64f42bf2ee20",
      "name": "User One",
    },
  },
}
`;
